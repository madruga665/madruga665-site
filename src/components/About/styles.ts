import styled from "styled-components";
import { AnimationPositionLeft, AnimationPositionRight } from "../../styles/animations";

export const AboutContainer = styled("section")`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 1.5rem 1rem;

  @media only screen and (min-width: 768px) {
    width: 90%;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    margin: 0;
  }
`;

export const ImageContainer = styled("div")`
  margin: 0 0 1rem 0;
  animation: ${AnimationPositionLeft} 2.5s ease-in-out;
  .about-image {
    border-radius: 2rem;
  }
`;

export const TextContainer = styled("div")`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 0.5rem;
  animation: ${AnimationPositionRight} 2.5s ease-in-out;
  & p {
    font-size: 1.5rem;
    text-align: center;
    color: #fff;
    max-width: 30rem;

    @media only screen and (min-width: 768px) {
      margin-left: 1rem;
      text-align: left;
    }
  }
`;
