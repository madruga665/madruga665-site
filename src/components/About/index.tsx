import { AboutContainer, ImageContainer, TextContainer } from "./styles";

const About = () => {
  return (
    <AboutContainer>
      <ImageContainer>
        <img
          className='about-image'
          src='/assets/madruga-image-about.jpg'
          alt='Luciano Amancio'
          width={300}
          height={400}
        />
      </ImageContainer>
      <TextContainer>
        <p>
          Olá, me chamo Luciano Amâncio, tenho 34 anos e moro em Guarujá-SP. Sou pai da Julia do
          Heitor e do Dante. Sou Desenvolvedor Web Full Stack, formado pela Trybe e atualmente
          trabalho na Remessa Online Criei esse site para mostrar um pouco de minhas habilidades e
          experiências.
        </p>
      </TextContainer>
    </AboutContainer>
  );
};

export default About;
