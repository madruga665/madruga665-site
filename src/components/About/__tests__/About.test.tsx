import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import About from "..";

describe("Testes do componente About", () => {
  test("Snapshot do componente", () => {
    const tree = renderer.create(<About />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  test("A imagem esta sendo renderizado", () => {
    render(<About />);
    const madrugaImage = screen.getByRole("img", {
      name: /luciano amancio/i,
    });
    expect(madrugaImage).toBeInTheDocument();
  });
  test("O texto 'sobre' está sendo renderizado", () => {
    render(<About />);
    const aboutText = screen.getByText(
      /olá, me chamo , tenho 34 anos e moro em guarujá\-sp\.sou pai da do e do \.sou desenvolvedor web full stack, formado pela e atualmente trabalho na criei esse site para mostrar um pouco de minhas habilidades e experiências \./i
    );
    expect(aboutText).toBeInTheDocument();
  });
});
