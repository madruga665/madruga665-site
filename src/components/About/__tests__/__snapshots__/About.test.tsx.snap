// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testes do componente About Snapshot do componente 1`] = `
<section
  className="sc-bdvvtL giuIBI"
>
  <div
    className="sc-gsDKAQ bDeHbX"
  >
    <img
      alt="Luciano Amancio"
      className="about-image"
      height={400}
      src="/assets/madruga-image-about.jpg"
      width={300}
    />
  </div>
  <div
    className="sc-dkPtRN bYqQBK"
  >
    <p>
      Olá, me chamo 
      <strong>
        Luciano Amâncio
      </strong>
      , tenho 34 anos e moro em Guarujá-SP.
      <br />
      Sou pai da 
      <strong>
        Julia
      </strong>
       do 
      <strong>
        Heitor
      </strong>
       e do 
      <strong>
        Dante
      </strong>
      .
      <br />
      Sou Desenvolvedor Web Full Stack, formado pela 
      <strong>
        Trybe
      </strong>
       e atualmente trabalho na 
      <strong>
        Remessa Online
      </strong>
      <br />
      Criei esse site para mostrar um pouco de minhas habilidades e experiências .
    </p>
  </div>
</section>
`;
